### ISSUE TITLE
Amplify version 2.12, EventRecorder causes crashed app issue when submit event

### ISSUE THREAD

**Issue by santoso8118** [aws-amplify/amplify-swift#3328]
### Describe the bug

My app is using Amplify version 2.12 and got crashed app.

### Steps To Reproduce

```swift
We don't know how to reproduce the issue
```

### Expected behavior

App should not be crashed

### Amplify Framework Version

2.12.0

### Amplify Categories

Analytics

### Dependency manager

Swift PM

### Swift version

Swift 5.7

### CLI version

2.12.0

### Xcode version

14.2

### Relevant log output

```shell
<details>
<summary>EventRecorder.submit(pinpointEvents:endpointProfile:)</summary>


Crashed: com.apple.root.background-qos.cooperative
0  libswiftCore.dylib             0x3bd600 _swift_release_dealloc + 32
1  libswiftCore.dylib             0x3be43c bool swift::RefCounts<swift::RefCountBitsT<(swift::RefCountInlinedness)1> >::doDecrementSlow<(swift::PerformDeinit)1>(swift::RefCountBitsT<(swift::RefCountInlinedness)1>, unsigned int) + 132
2  MyApp                         0x20510e0 specialized _ArrayBuffer._consumeAndCreateNew(bufferIsUnique:minimumCapacity:growForAppend:) + 4364259552 (<compiler-generated>:4364259552)
3  MyApp                         0x2033c48 EventRecorder.submit(pinpointEvents:endpointProfile:) + 266 (EventRecorder.swift:266)
4  libswift_Concurrency.dylib     0x41948 swift::runJobInEstablishedExecutorContext(swift::Job*) + 416
5  libswift_Concurrency.dylib     0x42868 swift_job_runImpl(swift::Job*, swift::ExecutorRef) + 72
6  libdispatch.dylib              0x15944 _dispatch_root_queue_drain + 396
7  libdispatch.dylib              0x16158 _dispatch_worker_thread2 + 164
8  libsystem_pthread.dylib        0xda0 _pthread_wqthread + 228
9  libsystem_pthread.dylib        0xb7c start_wqthread + 8
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

iOS 16.6.1

### Device

iPhone 12 Pro Max

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by phantumcode** [comment]
@santoso8118 Thank you for submitting your issue.  We'll investigate and try to reproduce the issue.  Do you have any sample code or sample app that can help us reproduce this issue?

**Comment by ruisebas** [comment]
@santoso8118 could you also please share the full symbolicated crash log? 
it contains additional information that might help us identify the root cause of the crash

**Comment by ruisebas** [comment]
I was able to reproduce this when `Amplify.Analytics.flushEvents()` is called while a previous event submission is still ongoing. This could happen due to different factors, such as: many events in the local database, slow connectivity, small auto flush interval, or a combination of them.

In any case, I'm working on a fix.

**Comment by santoso8118** [comment]
Hi @ruisebas 

Thanks for your feedback !

If you are able to reproduce the issue could you tell me what is the steps ?

This is the full crashed log. Please check and let me know if you need anything else to investigate the issue.

[crashlog.txt](https://github.com/aws-amplify/amplify-swift/files/13184817/crashlog.txt)

**Comment by santoso8118** [comment]
Hi @ruisebas :
I got another crashed app issues related to EventRecorder
This is the crashed log:

This is the key:
SQLite/Statement.swift:211: Fatal error: 'try!' expression unexpectedly raised an error: disk I/O error (code: 10)

Full log:

[crashedlog2.txt](https://github.com/aws-amplify/amplify-swift/files/13200321/crashedlog2.txt)

Please let me know if any update for the fixes?

**Comment by ruisebas** [comment]
These issues seem to be caused by race conditions.
The fix has been merged and will be released in the next couple of days.

**Comment by ruisebas** [comment]
Released in [2.21.4](https://github.com/aws-amplify/amplify-swift/releases/tag/2.21.4)
